### PR TITLE
Update `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by g-votte**
## Motivation
- Currently, the information for contributors is cluttered between `CONTRIBUTING.md` and [Wiki](https://github.com/optuna/optuna/wiki). This PR recompiles those pieces of information referencing them from `CONTRIBUTING.md`.
- Currently, some parts of the section structure of `CONTRIBUTING.md` is asymmetry. For example, `Unit Tests` and `Documentation` are placed in different layer despite that both subsections are guidelines to create pull request. This PR restructures them.

## Description of the changes
- Reference the conventions and policies in Optuna Wiki
- Restructure the sections as:
    - Contribution Guidelines
    - Pull Request Guidelines
        - Setup Optuna
        - Checking the Format, Coding Style, and Typle Hinsts
        - Documentation
        - Unit Tests
        - Continuous Integration and Local Verification
        - Creating a Pull Request
    - Learning Optuna Implementation

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
@contramundum53 @nzw0301 Could you review this PR, please? If you are busy, please feel free to remove the assignment.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,15 +26,23 @@ If you feel like giving a hand, here are some ways:
     - __Contribution to Optuna includes not only sending pull requests, but also writing down your comments on issues and pull requests by others, and joining conversations/discussions on [Gitter](https://gitter.im/optuna/optuna).__
     - Also, sharing how you enjoy Optuna is a huge contribution! If you write a blog, let us know about it!
 
-If you write code, we have some conventions as follows.
 
-- [Guidelines](#guidelines)
+# Pull Request Guidelines
```
Comment (contramundum53) [inline CONTRIBUTING.md:30]:
```suggestion
## Pull Request Guidelines
```
The first-level header is already used for title.

**Comment by contramundum53** [comment]
Isn't `Pull Request` part of `Contribution`?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -26,15 +26,23 @@ If you feel like giving a hand, here are some ways:
...
-- [Learning Optuna's Implementation](#learning-optunas-implementation)
 
-## Guidelines
+Detailed conventions and policies to write, test, and maintain Optuna code are described in the [Optuna Wiki](https://github.com/optuna/optuna/wiki).
+
+- [Coding Style Conventions](https://github.com/optuna/optuna/wiki/Coding-Style-Conventions)
+- [Deprecation Policy](https://github.com/optuna/optuna/wiki/Deprecation-policy)
+- [Test Policy](https://github.com/optuna/optuna/wiki/Test-Policy)
 
 ### Setup Optuna
```
Comment (contramundum53) [inline CONTRIBUTING.md:47]:
```suggestion
## Setup Optuna
```

**Comment by nzw0301** [comment]
I've been working a little bit too long this month, so let me unassign myself once.

**Comment by g-votte** [comment]
@contramundum53 

Thanks for your comment. I changed the structure again as:
- Contribution Guidelines
  - Pull Request Guidelines
    - Setup Optuna
    - Checking the Format, Coding Style, and Typle Hinsts
    - Documentation
    - Unit Tests
    - Continuous Integration and Local Verification
    - Creating a Pull Request
  - Learning Optuna Implementation

In addition, I removed the subsections of `title`, `example`,  and `description` in `Creating a Pull Request` because they've been too much nested so that the font size is almost the same as normal text.

File: CONTRIBUTING.md (hunk 2 above)
Comment (g-votte) [reply CONTRIBUTING.md:47]: Thanks for pointing this out. After [this change](https://github.com/optuna/optuna/pull/3726#issuecomment-1169485567), "Setup Optuna" is actually the 3rd level. Thus, I'll leave this line as it is.

**Review by contramundum53 — APPROVED** [review]
LGTM!

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**